File: backend-nestjs/src/modules/department-management/application/dto/department.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class DepartmentDto {
  @ApiProperty({ description: 'Department ID' })
  id: string;

  @ApiProperty({ description: 'Department name' })
  name: string;

  @ApiProperty({ description: 'Department code' })
  code: string;

  @ApiProperty({ description: 'Department description', nullable: true })
  description: string | null;

  @ApiProperty({ description: 'Is active' })
  isActive: boolean;

  @ApiProperty({ description: 'Created at' })
  createdAt: Date;

  @ApiProperty({ description: 'Updated at' })
  updatedAt: Date;

  constructor(partial: Partial<DepartmentDto>) {
    Object.assign(this, partial);
  }
}

File: backend/spring/feature-architecture/full/src/main/java/com/apptemplate/api/features/departments/DepartmentDto.java
package com.apptemplate.api.features.departments;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DepartmentDto {
    private Long id;
    private String code;
    private String name;
    private String description;
    @JsonProperty("isActive")
    private boolean isActive;
    private long userCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public static DepartmentDto fromEntity(Department department, long userCount) {
        return DepartmentDto.builder()
                .id(department.getId())
                .code(department.getCode())
                .name(department.getName())
                .description(department.getDescription())
                .isActive(department.isActive())
                .userCount(userCount)
                .createdAt(department.getCreatedAt())
                .updatedAt(department.getUpdatedAt())
                .build();
    }
}
